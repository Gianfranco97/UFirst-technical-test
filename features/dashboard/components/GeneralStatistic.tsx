import { Card, Statistic, Row, Col } from 'antd';

type Props = {
  totalRequest?: number,
  totalInvalidRequest?: number,
  totalRequestPerMinute?: number
};

function GeneralStatistic({ totalRequest, totalInvalidRequest, totalRequestPerMinute }: Props) {
  return (
    <Card title="General statistic">
      <Row gutter={[8, 24]}>
        <Col span={8}>
          <Statistic
            title="Total request"
            value={totalRequest}
            loading={!totalRequest}
          />
        </Col>

        <Col span={8}>
          <Statistic
            title="Total request per minute"
            value={totalRequestPerMinute}
            loading={!totalRequestPerMinute}
          />
        </Col>

        <Col span={8}>
          <Statistic
            title="Invalid request"
            value={totalInvalidRequest}
            loading={!totalInvalidRequest}
          />
        </Col>
      </Row>
    </Card>
  );
};

export default GeneralStatistic;
