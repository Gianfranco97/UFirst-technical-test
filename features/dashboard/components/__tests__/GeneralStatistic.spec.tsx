import { expect } from '@jest/globals';
import { render } from '@testing-library/react'
import GeneralStatistic from '../GeneralStatistic';

describe('GeneralStatistic', () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      }))
    });
  });

  it('renders correctly', () => {
    const { container } = render(
      <GeneralStatistic
        totalRequest={10}
        totalInvalidRequest={2}
        totalRequestPerMinute={4}
      />
    )

    expect(container).toMatchSnapshot()
  })
});