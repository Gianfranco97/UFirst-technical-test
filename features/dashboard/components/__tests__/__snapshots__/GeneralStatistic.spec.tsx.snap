// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralStatistic renders correctly 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          General statistic
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-row"
        style="margin: -12px -4px -12px -4px;"
      >
        <div
          class="ant-col ant-col-8"
          style="padding: 12px 4px 12px 4px;"
        >
          <div
            class="ant-statistic"
          >
            <div
              class="ant-statistic-title"
            >
              Total request
            </div>
            <div
              class="ant-statistic-content"
            >
              <span
                class="ant-statistic-content-value"
              >
                <span
                  class="ant-statistic-content-value-int"
                >
                  10
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-col ant-col-8"
          style="padding: 12px 4px 12px 4px;"
        >
          <div
            class="ant-statistic"
          >
            <div
              class="ant-statistic-title"
            >
              Total request per minute
            </div>
            <div
              class="ant-statistic-content"
            >
              <span
                class="ant-statistic-content-value"
              >
                <span
                  class="ant-statistic-content-value-int"
                >
                  4
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-col ant-col-8"
          style="padding: 12px 4px 12px 4px;"
        >
          <div
            class="ant-statistic"
          >
            <div
              class="ant-statistic-title"
            >
              Invalid request
            </div>
            <div
              class="ant-statistic-content"
            >
              <span
                class="ant-statistic-content-value"
              >
                <span
                  class="ant-statistic-content-value-int"
                >
                  2
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
